src/averageAges.js

```diff
@@ -52,7 +58,15 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  return +people
+    .filter(child =>
+      people.some(mother => onlyWithSon ? child.mother === mother.name
+        && child.sex === 'm'
+        : child.mother === mother.name))
+    .map(person => person.born - people.find(pers =>
+      pers.name === person.mother).born)
+    .reduce((prev, age, i, arr) => prev + age / arr.length, 0)
```

Comment:
```suggestion
    .reduce((prev, age,, arr) => prev + age / arr.length, 0)
```
Якщо 'i' не використовується, можна його не писати, а просто поставити кому